### PR TITLE
Feature/actions tests

### PR DESCRIPTION
**Pull request by vdubr**
action tests uses for common actions common tests
refactoring some store/actions into new structure
list of tests/missing tests in /tests/state/README.md

### CODE REVIEW

File: tests/state/_common/actions/setActiveKey-test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import {commonActionTypesObj as actionTypes} from '../../../constants';
+
+const tests = [
+	{
+		name: 'It dispatch setActiveKey.',
+		action: (actions, actionTypes, options) => {
+			return (dispatch, getState) => {
+				let action;
+				// for common testing
+				if (actionTypes && options) {
```
Comment (vlach1989) [inline tests/state/_common/actions/setActiveKey-test.js:12]: Is options check necessary?

File: tests/state/helpers.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * Returns empty state
+ * @returns {Object}
+ */
+const defaultGetState = () => ({});
+
+/**
+ * Run set of actions tests in scope of parameters like store, dataType, categoryPath.
+ * @param {String} dataType String identificator of store. It is also used in request url like `/rest/${categoryPath}/filtered/${dataType}`.
```
Comment (vlach1989) [inline tests/state/helpers.js:13]:
```suggestion
 * @param {String} dataType String identifier of store. It is also used in request url like `/rest/${categoryPath}/filtered/${dataType}`.
```

File: tests/constants.js [hunk 3]
```diff
@@ -36,49 +37,112 @@ export const expectedLayerTreesSelectors = [...baseMetadataSelectors];
 export const expectedStylesSelectors = [...baseMetadataSelectors];
 export const expectedViewsSelectors = [...baseMetadataSelectors];
 
+export function getDeepValues(obj) {
```
Comment (vlach1989) [inline tests/constants.js:40]: either don't export it or move it from constants

File: tests/state/Cases/actions-test.js [hunk 4]
```diff
@@ -437,124 +437,124 @@ describe('state/Cases/actions', function () {
...
-	});
-
-	it('useKeysClear', function () {
-		actions.useKeysClear('cid')(dispatch);
-
-		assert.deepStrictEqual(dispatchedActions, [
-			{componentId: 'cid', type: 'CASES.USE.KEYS.CLEAR'},
-		]);
-	});
+	// it('useIndexed', function () {
```
Comment (vlach1989) [inline tests/state/Cases/actions-test.js:440]: deprecated tests could be removed

File: tests/store.js [hunk 5]
```diff
@@ -1,10 +1,22 @@
+/**
+ * Create clear set of helper functions for testing dispatche actions on store.
```
Comment (vlach1989) [inline tests/store.js:2]:
```suggestion
 * Create clear set of helper functions for testing dispatched actions on store.
```

File: tests/state/_common/actions/useIndexed-test.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
+import {assert} from 'chai';
+import slash from 'slash';
+import commonActions from '../../../../src/state/_common/actions';
+import testBatchRunner from '../../helpers';
+import {commonActionTypesObj as actionTypes} from '../../../constants';
+
+const tests = [
+	{
+		name: 'It register index and loads indexed data.',
```
Comment (vlach1989) [inline tests/state/_common/actions/useIndexed-test.js:9]: What about the case when the data are already indexed?

File: tests/state/_common/actions-test.js [hunk 7]
```diff
@@ -353,78 +353,78 @@ describe('state/_common/actions', function () {
...
-						data: [
-							{
-								key: 'k1',
-							},
-						],
-					},
-				]);
-			});
-	});
+	// it('ensureKeys', function () {
```
Comment (vlach1989) [inline tests/state/_common/actions-test.js:356]: deprecated tests can be removed

**Review by vlach1989 — CHANGES_REQUESTED** [review]

File: tests/constants.js (hunk 3 above)
Comment (vdubr) [reply tests/constants.js]: export removed

File: tests/store.js (hunk 5 above)
Comment (vdubr) [reply tests/store.js]: fixed

File: tests/state/_common/actions-test.js (hunk 7 above)
Comment (vdubr) [reply tests/state/_common/actions-test.js]: removed file

File: tests/state/_common/actions/setActiveKey-test.js (hunk 1 above)
Comment (vdubr) [reply tests/state/_common/actions/setActiveKey-test.js:12]: Yes, existence of actionTypes and options means that tests is for common action not implemented in some store.

File: tests/state/Cases/actions-test.js (hunk 4 above)
Comment (vdubr) [reply tests/state/Cases/actions-test.js]: Removed

File: src/constants/ActionTypes.js [hunk 8]
```diff
@@ -541,6 +543,7 @@ export default utils.deepKeyMirror({
 			REQUEST: null,
 		},
 		SET_ACTIVE_KEY: null,
+		SET_ACTIVE_KEYS: null,
```
Comment (vlach1989) [inline src/constants/ActionTypes.js:546]: Does it make sense to set multiple scopes as active?

File: src/constants/ActionTypes.js [hunk 9]
```diff
@@ -730,6 +743,7 @@ export default utils.deepKeyMirror({
...
```
Comment (vlach1989) [inline src/constants/ActionTypes.js:746]: Does it make sense to set multiple views as active?

File: tests/state/Areas/AreaTrees/actions/setActiveKey-test.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+];
+
+const dataType = 'areaTree';
+const categoryPath = 'metadata';
+describe(
+	'state/Areas/AreaTrees/actions',
+	testBatchRunner(dataType, categoryPath, tests, actions, actionTypes)
+);
+
+export default tests;
```
Comment (vlach1989) [inline tests/state/Areas/AreaTrees/actions/setActiveKey-test.js:22]: I'm confused. setActiveKey is part of a common test set, so why is it here specifically?

File: tests/state/_common/actions/action-test.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
+import {assert} from 'chai';
+import commonActions from '../../../../src/state/_common/actions';
+import testBatchRunner from '../../helpers';
+import {commonActionTypesObj as actionTypes} from '../../../constants';
+
+const tests = [
+	{
+		name: 'It fail on unexistinf action in actionTypes.',
```
Comment (vlach1989) [inline tests/state/_common/actions/action-test.js:8]:
```suggestion
		name: 'It fail on unexisting action in actionTypes.',
```

File: src/constants/ActionTypes.js (hunk 8 above)
Comment (vdubr) [reply src/constants/ActionTypes.js:546]: Not much for me. Said. This is result of merging setActiveKey and setActiveKeys into one class SETTING_ACTIVE_ACTIONS. And yes, I think we should split it into two classes: SETTING_ACTIVE_KEYS_ACTIONS and SETTING_ACTIVE_KEY_ACTIONS where SETTING_ACTIVE_KEYS_ACTIONS will contain both actions setActiveKey and setActiveKeys.

File: tests/state/Areas/AreaTrees/actions/setActiveKey-test.js (hunk 10 above)
Comment (vdubr) [reply tests/state/Areas/AreaTrees/actions/setActiveKey-test.js]: I would like to answer something reasonable, but it was done at the end of the shift.  Fixed: d51baa0f7d34d195a68bdf89ab1a96bc693e7205

File: tests/state/_common/actions/useIndexed-test.js (hunk 6 above)
Comment (vdubr) [reply tests/state/_common/actions/useIndexed-test.js:9]: useIndexed only wraps "ensureIndexed" and add registering of the index. ensureIndexed-test solve case when data are already loaded.

File: src/constants/ActionTypes.js (hunk 9 above)
Comment (vdubr) [reply src/constants/ActionTypes.js:746]: no, fixed

**Review by vlach1989 — APPROVED** [review]

**Review by tmdmc — APPROVED** [review]
tl;dr

**Pull request merged**